File: G-for-g-Practice/Array leaders.py
class Solution:
    def leaders(self, arr):
        # code here
        max_From_Right= float('-inf')
        result = []
        
        for i in range(len(arr)-1, -1, -1):
            if arr[i]>=max_From_Right:
                result.append(arr[i])
                max_From_Right = arr[i]
        
        return result[::-1]
            

File: G-for-g-Practice/Array leaders.java


class Solution {
    static ArrayList<Integer> leaders(int arr[]) {
        // code here
        LinkedList<Integer> result = new LinkedList<>();
        int maxFromRight = Integer.MIN_VALUE;
        
        
        for(int i = arr.length-1; i>=0; i--){
            if (arr[i] >= maxFromRight){
                result.addFirst(arr[i]);
                maxFromRight = arr[i];
            }
        }

        return new ArrayList<>(result);
    }
}
